### PR TITLE
feat: 🎸 cash flow statement guide for crypto exchange

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP
- src/km/km-20230830001.md

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component:  0
- new loop: 0
- high risk: 0
- new sql: 0

### CODE REVIEW

File: src/km/km-20230825001.md [hunk 1]
```diff
@@ -0,0 +1,737 @@
+---
+date: 1692953015
+title: '撰寫加密貨幣交易所現金流量表指南'
+description: '本文將介紹現金流量表的本質，以及製作現金流量表的方法，接著一一用直接法呈現加密貨幣交易所的營運活動造成的現金流量表變化，期望達到示範作用並引起更多加密貨幣交易所監管相關的討論。'
+picture: '/km/cash-flow-statements-img.jpg'
+category: ['KM_CATEGORY.FINANCIALS', 'KM_CATEGORY.CASH_FLOW_STATEMENT']
+authorId: 'shirley'
+---
+
+# 加密貨幣交易所的現金流量表撰寫指南
```
Comment (arealclimber) [inline src/km/km-20230825001.md:10]: - 給供應商的費用，例如手續費
- 掮客反傭，要給用戶的分潤

Comment (arealclimber) [reply src/km/km-20230825001.md:10]: 條列各個行為跟對應的科目，如USDT received from customers

Comment (arealclimber) [reply src/km/km-20230825001.md:10]: 重複的USDT received from customers: 從用戶那邊賺到的 USDT

File: src/km/km-20230825001.md [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+
+Cash paid to customers: ($8)
+
+- 保證金為涵蓋點差的價格除以槓桿
+- 多單
+  - 保證金為 5x1000x1.01/5=$1010，((800-1000)/1000)x5=-1 觸及強制平倉價，損失所有保證金 1010 美元
+  - 強制平倉價格 1000x(1-0.2)x1.0001=800.08，強制平倉手續費 0.08 美元，由交易所執行強制平倉，由保險
+    基金吸收強制平倉手續費。
+- 空單
+  - 保證金 8x1000x0.99/5=$1584，(-(800x1.01-1000))x9=1344
```
Comment (arealclimber) [inline src/km/km-20230825001.md:549]: 7*1000*....=1386

File: src/km/km-20230825001.md [hunk 3]
```diff
@@ -0,0 +1,737 @@
...
+### 過程
+
+- 保證金為涵蓋點差的價格除以槓桿
+- 多單
+  - 保證金為 5x1000x1.01/5=$1010，((800-1000)/1000)x5=-1 觸及強制平倉價，損失所有保證金 1010 美元
+  - 強制平倉價格 1000x(1-0.2)x1.0001=800.08，強制平倉手續費 0.08 美元，由交易所執行強制平倉，由保險
+    基金吸收強制平倉手續費。
+- 空單
+
+  - 保證金 1x1000x0.99/5=$198，(-(800x1.01-1000))x2=192
```
Comment (arealclimber) [inline src/km/km-20230825001.md:592]: 保證金 1x1000x0.99/5=$198，(-(800x1.01-1000))x`1`=192

File: src/km/km-20230825001.md [hunk 4]
```diff
@@ -0,0 +1,737 @@
...
+
+USDT paid to customers: ($8)
+
+- 保證金為涵蓋點差的價格除以槓桿
+- 多單
+  - 保證金為 5x1000x1.01/5=$1010，((800-1000)/1000)x5=-1 觸及強制平倉價，損失所有保證金 1010 美元
+  - 強制平倉價格 1000x(1-0.2)x1.0001=800.08，強制平倉手續費 0.08 美元，由交易所執行強制平倉，由保險
+    基金吸收強制平倉手續費。
+- 空單
+  - 保證金 8x1000x0.99/5=$1584，(-(800x1.01-1000))x9=1344
```
Comment (arealclimber) [inline src/km/km-20230825001.md:660]: 檢查數字

File: src/km/km-20230825001.md [hunk 5]
```diff
@@ -0,0 +1,737 @@
...
+    - 空單關倉時用戶拿回保證金 1980 美元跟獲利 1344 美元
+
+### 結果
```
Comment (arealclimber) [inline src/km/km-20230825001.md:663]: 將 以清算 都改成 "已關倉"

**Comment by arealclimber** [comment]
https://github.com/CAFECA-IO/BAIFA/issues/185
給供應商的費用，例如手續費
掮客反傭，要給用戶的分潤

https://github.com/CAFECA-IO/BAIFA/issues/186
條列各個行為跟對應的科目，如USDT received from customers

File: src/km/km-20230825001.md [hunk 6]
```diff
@@ -0,0 +1,880 @@
...
+
+Cryptocurrency inflows: $6000
+
+Disposal of cryptocurrencies with non-cash consideration: ($6006)
+
+Insurance fund for perpetual contracts with non-cash consideration: $9.6
+
+Cryptocurrencies paid to customers for profits in perpetual contract: ($1357)
+
+Cryptocurrencies received from customers as transaction feess: $3.42
```
Comment (arealclimber) [inline src/km/km-20230825001.md:868]: feess

File: src/km/km-20230825001.md [hunk 7]
```diff
@@ -0,0 +1,880 @@
...
+| Insurance fund for perpetual contracts with non-cash consideration             | 為了保證永續合約正常運作的基金，以加密貨幣形式 |
+| Cryptocurrencies received from customers for liquidation in perpetual contract | 從客戶收到的用於永續合約交易清算的加密貨幣     |
+| Cryptocurrencies paid to customers for perpetual contract profits              | 用加密貨幣支付給用戶的永續合約利潤             |
+| Cryptocurrencies paid to customers for funding rates in perpetual contract     | 用加密貨幣支付給用戶的永續合約資金費率         |
+| Cryptocurrencies paid to customers for profits in perpetual contract           | 用加密貨幣支付給客戶的永續合約交易利潤         |
+
+## 其他補充項目
+
+- 與現金有關
+  - Effect of exchange rates on cash, cash equivalents, and restricted cash 其餘法幣兌換成功能性貨幣
```
Comment (arealclimber) [inline src/km/km-20230825001.md:107]: 去掉 未實現損益的部分

File: src/km/km-20230825001.md [hunk 8]
```diff
@@ -0,0 +1,880 @@
...
+---
+
+### 交易所跟用戶都用 USDT 交易，部分倉位未關倉，交易所收手續費並有返傭機制，以及與供應商來往的例子
+
+- 指定期間內，交易所本身資產(狀態)有 10 BTC，10,000 USDT ；用戶總共開 12 BTC 的多單，2 BTC 的空單，
+  剩餘 5 BTC 的多單跟 1 BTC 的空單未關倉，交易所在這期間買了 5 BTC - 資金費率始終為 0.1%，交易所買
+  BTC 的匯率為 1200 USD/BTC，用戶開多單的匯率為 800 USD/BTC、在 1000 USD/BTC 關倉，用戶開空單的匯率
+  為 900 USD/BTC、在 800 USD/BTC 關倉；財報結算時匯率 800 USDT/BTC， 交易所點差固定為 2 %，單向為
+  (2%/2=0.01)，1 USDT = 1 USD，強制平倉手續費為 0.01%，taker 及 maker 手續費皆為 0.03%，無論開倉關倉
+  都會收手續費； 其中返傭比例為手續費 30%；需定期支付供應商一筆費用，可能是固定的或浮動的，在此假設
```
Comment (arealclimber) [inline src/km/km-20230825001.md:802]: 換行

File: src/km/km-20230830001.md [hunk 9]
```diff
@@ -0,0 +1,983 @@
...
+| USDT                                                                          | 0              | 0              | —                   | 0             | 0              | —                   |
+| Total cryptocurrencies received from customers as transaction fees            | —              | $0             | —                   | —             | $0             | —                   |
+|                                                                               |                |                |                     |               |                |                     |
+|                                                                               |                |                |                     |               |                |                     |
+|                                                                               |                |                |                     |               |                |                     |
+| Cryptocurrencies received from customers for liquidation in CFD trading       | 7 月. 30, 2023 |                |                     | 7 月. 1, 2023 |                |                     |
```
Comment (arealclimber) [inline src/km/km-20230830001.md:211]: 將日期改成英文

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**